examples/static-spark-encrypted/kms.tf

```diff
@@ -0,0 +1,82 @@
+locals {
+  effective_kms_key_arn = var.kms_key_arn == "" ? aws_kms_key.kms_encryption_key[0].arn : var.kms_key_arn
+  account_id            = data.aws_caller_identity.current.account_id
+}
+
+data "aws_caller_identity" "current" {}
+
+//The minimal policy document for the KMS key
+data "aws_iam_policy_document" "kms_key" {
+  version = "2012-10-17"
+
+  statement {
+    sid       = "Enable IAM User Permissions"
+    effect    = "Allow"
+    actions   = ["kms:*"]
+    resources = ["*"]
+
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${local.account_id}:root"]
+    }
+  }
+
+  statement {
+    sid    = "Allow use of the key"
+    effect = "Allow"
+    principals {
+      type = "AWS"
+      identifiers = [
+        "${module.emr-spark.emr_ec2_role_arn}",
+        "${module.emr-spark.emr_service_role_arn}"
+      ]
+    }
+    actions = [
+      "kms:Encrypt",
+      "kms:Decrypt",
+      "kms:ReEncrypt*",
+      "kms:GenerateDataKey*",
+      "kms:DescribeKey"
+    ]
+    resources = ["*"]
```

Comment:
Can we narrow this down?

Reply:
We confirmed in a call that this is the way.